### PR TITLE
Limit bullet overlays on walls

### PR DESCRIPTION
**Pull request by AnturK**
Badcode

### CODE REVIEW

File: code/game/turfs/simulated/walls.dm [hunk 1]
```diff
@@ -297,10 +299,12 @@
...
 /turf/closed/wall/proc/add_dent(denttype, x=rand(-8, 8), y=rand(-8, 8))
-	var/mutable_appearance/decal = pick(dent_decal_list[denttype])
-	decal.pixel_x = x
-	decal.pixel_y = y
-
-	cut_overlay(dent_decals)
-	LAZYADD(dent_decals, decal)
-	add_overlay(dent_decals)
+	if(LAZYLEN(dent_decals) < MAX_DENT_DECALS)
+		var/mutable_appearance/decal = pick(dent_decal_list[denttype])
```
Comment (Cyberboss) [inline code/game/turfs/simulated/walls.dm]: Early return

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by nicbn** [comment]
Call undef at the end of the file, and raise this limit.

**Comment by vuonojenmustaturska** [comment]
5 is plenty

**Review by Cyberboss — DISMISSED** [review]

**Comment by Iatots** [comment]
Badcoder.
An intense firefight ending with all walls bearing the exact same number of holes kill immersion real quick.
A wall starts to become more holes than panels at around 42 shots (assuming no overlap) so maybe start around there for the limit.

**Comment by Buggy123** [comment]
Agreed, increase the bullet hole limit significantly at the very least. I think this would only be useful to prevent abuse - such as somehow printing thousands of bullets and spamming enough holes to potentially cause lag.

**Comment by vuonojenmustaturska** [comment]
potentially cause lag? this is literally the cause of most server crashes right now

**Comment by Iatots** [comment]
Is the problem a high amount of holes on 1 wall or the global amount of decals?

**Comment by vuonojenmustaturska** [comment]
The former is the more immediate concern, but the latter is also important because the holes are only cleared at roundstart or when a singulo eats the station, and their cost is largely the same regardless of whether there are 40000 extra overlays on one object or across the station

**Comment by Cyberboss** [comment]
I think we can handle 10 per wall at least

**Comment by Arianya** [comment]
The actual solution is to make bulletholes act as tiny atmos leaks/passthroughs that can be repaired with a welding tool or a sheet of metal.

**Comment by vuonojenmustaturska** [comment]
Yes, the solution to a subsystem killing the server is to make it also stress another subsystem by making all tiles near a bullet hole permanently active atmos regions and remove the whole ATMOS_PASS_DENSITY optimization.

@duncathan

**Comment by Arianya** [comment]
Of course it would, because people would have a reason to fix dents, rather then leaving walls with 89 holes.

Regardless, the limit should be higher then 5 lmao

**Comment by AnturK** [comment]
I'll bump it to 15, we can amortize many overlays over many objects with CHECK_TICK, not so much single object having million overlays

**Review by Cyberboss — APPROVED** [review]

**Comment by duncathan** [comment]
lmao, bullet holes acting as atmos leaks, fuckin imagine
that's such a retarded idea I can't do anything but laugh

**Comment by nicbn** [comment]
>the holes are only cleared at roundstart or when a singulo eats the station

Just use a welding tool

A better solution is make the wall fall apart after some amount of shots

**Comment by RemieRichards** [comment]
@nicbn is 100% correct, there's no other acceptable answer, the wall should break at 'X' holes, and that is your arbitrary limit.

**Comment by Iatots** [comment]
It could break into sharp pieces too for when the bullets run out.

**Comment by Arianya** [comment]
@RemieRichards Lets please not have structural damage 2.0 where people find yet more ways to avoid the construction or ID systems in favour of brute dumb force.

**Comment by vuonojenmustaturska** [comment]
@RemieRichards the problem here is that a bullet hole does not mean that the wall is damaged in any way, only that at some point in time it collided with a projectile type. Reinforced walls for example should probably not be destroyed by laser tag guns, but making bullet holes destroy a wall will do that. 

@nicbn I can guarantee you with one hundred percent certainty that in no point in time will I ever take a 24 hour job hopping between the servers using a welding tool on walls that have been shot at when a better solution (this PR) exists.

**Pull request merged**